conan/tools/microsoft/msbuild.py

```diff
@@ -53,8 +53,7 @@ def command(self, sln):
 
     def build(self, sln):
         cmd = self.command(sln)
-        vcvars = os.path.join(self._conanfile.generators_folder, "conanvcvars")
-        self._conanfile.run(cmd, env=["conanbuildenv", vcvars])
+        self._conanfile.run(cmd)
```

Comment:
I don't see who is adding ``conanvcvars`` now

Reply:
The idea is to have a explicit `Vcvars(self).generate()` or `generators="Vcvars"` and it will be automatically added to the list of environments.

Reply:
I have a draft in a branch but with more things and very dirty yet.

Reply:
The problem with the vcvars generator is that it might be too coupled to other generator, like ``CMakeToolchain`` is the one that has the information to decide whether to generate ``vcvars`` or not. I don't see it, but if you have a branch we can check it later.

Reply:
Now the `write_conanvcvars()` function is adding the script to the list. We can add later (in other ticket/release) the generator for the user.